### PR TITLE
[data_release] Add new permission for viewing only

### PR DESCRIPTION
**Pull request by ridz1208**
## Brief summary of changes
This adds a new permission to the database to allow users view/download access to the data release module. the other 2 permissions that already exist give the user slightly more power, this one is the most basic access.

addresses https://github.com/aces/Loris/issues/5496

### CODE REVIEW

**Comment by ridz1208** [comment]
@driusan not sure if critical or not ?

**Comment by driusan** [comment]
At this point I think it's better to err on the side of adding the "Critical" tag if there's ambiguity.

File: modules/data_release/php/data_release.class.inc [hunk 1]
```diff
@@ -26,6 +26,23 @@ namespace LORIS\data_release;
...
+     *
+     * @return bool
+     */
+    function _hasAccess(\User $user) : bool
+    {
+        // check user permissions
+        return (
+            $user->hasPermission('data_release_view')
+            || $user->hasPermission('data_release_upload')
+            || $user->hasPermission('data_release_edit_file_access')
```
[the remark this answers is not in the document]
Comment (ridz1208) [reply modules/data_release/php/data_release.class.inc]: no prob

File: raisinbread/RB_files/RB_permissions.sql [hunk 2]
```diff
@@ -56,5 +56,6 @@ INSERT INTO `permissions` (`permID`, `code`, `description`, `categoryID`) VALUES
 INSERT INTO `permissions` (`permID`, `code`, `description`, `categoryID`) VALUES (53,'publication_view','Publication - Access to module',2);
 INSERT INTO `permissions` (`permID`, `code`, `description`, `categoryID`) VALUES (54,'publication_propose','Publication - Propose a project',2);
 INSERT INTO `permissions` (`permID`, `code`, `description`, `categoryID`) VALUES (55,'publication_approve','Publication - Approve or reject proposed publication projects',2);
+INSERT INTO `permissions` (`permID`, `code`, `description`, `categoryID`) VALUES (56,'data_release_view','Data Release: View releases',2);
```
Comment (johnsaigle) [inline raisinbread/RB_files/RB_permissions.sql]: The `permID` here doesn't match the patch above. Not sure if it matters.

File: SQL/New_patches/2019-11-01-Add_data_release_permissions.sql [hunk 3]
```diff
@@ -0,0 +1,8 @@
+INSERT INTO permissions (code,description,categoryID) VALUES 
+('data_release_view','Data Release: View releases',(SELECT ID FROM permissions_category WHERE Description='Permission'));
+
+INSERT INTO LorisMenuPermissions (MenuID, PermID)
+    SELECT m.ID, p.PermID FROM permissions p CROSS JOIN LorisMenu m WHERE p.code='data_release_view' AND m.Label='Data Release';
+
+INSERT IGNORE INTO user_perm_rel
+SELECT upr.userID, p.permID FROM user_perm_rel upr JOIN permissions p WHERE upr.permID=1 AND p.code IN ('data_release_view');
```
Comment (johnsaigle) [inline SQL/New_patches/2019-11-01-Add_data_release_permissions.sql]: Does this line give the permission to the first user in the SQL table aka admin?

Comment (ridz1208) [reply SQL/New_patches/2019-11-01-Add_data_release_permissions.sql]: yes

File: raisinbread/RB_files/RB_permissions.sql (hunk 2 above)
Comment (ridz1208) [reply raisinbread/RB_files/RB_permissions.sql]: no RB is meant to simulate a real project databse where the same thing would happen

**Pull request merged**